src/boards/include/boards/solderparty_rp2040_stamp.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+// -----------------------------------------------------
+// NOTE: THIS HEADER IS ALSO INCLUDED BY ASSEMBLER SO
+//       SHOULD ONLY CONSIST OF PREPROCESSOR DIRECTIVES
+// -----------------------------------------------------
+//
+//------------------------------------------------------------------------------------------
+// Board definition for the Solder Party RP2040 Stamp
+//
+// This header may be included by other board headers as "boards/solderparty_rp2040_stamp.h"
+
+#ifndef _BOARDS_SOLDERPARTY_RP2040_STAMP_H
+#define _BOARDS_SOLDERPARTY_RP2040_STAMP_H
+
+// For board detection
+#define SOLDERPARTY_RP2040_STAMP
+
+// --- UART ---
+#ifndef PICO_DEFAULT_UART
+#define PICO_DEFAULT_UART 0
+#endif
+#ifndef PICO_DEFAULT_UART_TX_PIN
+#define PICO_DEFAULT_UART_TX_PIN 0
+#endif
+#ifndef PICO_DEFAULT_UART_RX_PIN
+#define PICO_DEFAULT_UART_RX_PIN 1
+#endif
+
+// Red LED on the Carrier
+#ifndef PICO_DEFAULT_LED_PIN
+#define PICO_DEFAULT_LED_PIN   20
+#endif
+
+// Neopixel on the Stamp
```

Comment:
My version of this was:

```
// no PICO_DEFAULT_LED_PIN
// WS2812 can be jumpered out of circuit (JP1/JP2).
#ifndef PICO_NO_WS2812
#ifndef PICO_DEFAULT_WS2812_PIN
#define PICO_DEFAULT_WS2812_PIN 21
#endif
#endif // PICO_NO_WS2812
```

I decided not to set 'LED PIN' because it's only on the carrier, not the stamp itself.

Reply:
TBH most of these are specific to the Carrier, the Stamp itself can use any pin for any function (that the RP2040 allows), but the defaults are those that are used by the Carrier cause for many people that will be their entry point :)

Reply:
I would personally think that this board definition should be specific to the stamp on its own; if the carrier adds useful things to that then IMO it should be another board.

While I am aware that the stamp can use any pin, the point of these definitions is to provide sane defaults for the various examples and a good basis for building a project on top of it, so defining things which are not available unless you have a carrier board is not helpful.

Reply:
Yes, there should probably be a separate header for the carrier board that (pre) overrides anything it needs to, and includes the stamp header at the end to pick up the rest of the defaults